underscore.js

```diff
@@ -216,9 +213,8 @@
         accumulator = {};
       }
     }
-    if (obj == null) return accumulator;
     iteratee = optimizeCb(iteratee, context, 4);
-    var keys = obj.length !== +obj.length && _.keys(obj),
+    var keys = !isArrayLike(obj) && _.keys(obj),
```

Comment:
Intentional call, same thing going on in [reduce](https://github.com/jashkenas/underscore/blob/master/underscore.js#L186), map, some and others